examples/gcoap/gcoap_cli.c

```diff
@@ -314,12 +366,25 @@ int gcoap_cli_cmd(int argc, char **argv)
      */
     if (((argc == apos + 3) && (code_pos == 0)) ||
         ((argc == apos + 4) && (code_pos != 0))) {
-        gcoap_req_init(&pdu, &buf[0], CONFIG_GCOAP_PDU_BUF_SIZE, code_pos+1, argv[apos+2]);
+
+        char *uri = argv[apos+2];
+        int uri_len = strlen(argv[apos+2]);
+
+        if (_proxied) {
+            uri_len = snprintf(proxy_uri, 64, "coap://[%s]:%s%s", argv[apos], argv[apos+1], uri);
+            uri = proxy_uri;
+
+            gcoap_req_init(&pdu, &buf[0], CONFIG_GCOAP_PDU_BUF_SIZE, code_pos+1, NULL);
+            coap_opt_add_proxy_uri(&pdu, uri);
```

Comment:
Addition of this option here means the that setting the Content-Format option when there is a payload (`paylen > 0`) on l.392 will fail.

Reply:
hmm, I could move the addition of the proxy-uri option down .. I initially wanted to keep the changes close to the creation of the request .. doesn't seem to be possible, though.

Reply:
Yes, I would move it just above the call to `coap_opt_finish()`. Options must be added in order.

Reply:
I pushed two new fixups, one moves the call to `coap_opt_add_proxy_uri()` down, and the other one adjusts the `gcoap_req_init()` for block-wise transfer. I was quite simple to enable proxied requests for block transfers ..

>You could modify that to work with proxies, but IMO it adds too much to the example.

Should we keep it like the current diff, or should I remove the proxy operation from blockwise transfer completely?

Reply:
The current diff is simple enough and adds Block2 request support for proxies. That's great! Thanks for thinking it through.